pullRequestHandler.go

```diff
@@ -124,6 +129,103 @@ func hasUnsigned(req types.PullRequestOuter, client *github.Client) (bool, error
 	return hasUnsigned, err
 }
 
+func passBranchValidation(ctx context.Context, req types.PullRequestOuter, client *github.Client) bool {
+	isValidPullRequest := validateHeadBranch(ctx, req, client) && validateBaseBranch(ctx, req, client)
+
+	if !isValidPullRequest {
+		closePullRequest(req)
+	}
+
+	return isValidPullRequest
+}
+
+func validateHeadBranch(ctx context.Context, req types.PullRequestOuter, client *github.Client) bool {
+	if isMasterHeadBranch(req) {
+		body := "Thank you for your contribution. It appears that you are submitting changes directly from your master branch." +
+			"Please raise a new pull request from a named branch i.e. `git checkout -b my_feature`. "
+
+		comment(ctx, body, req, client)
+
+		applyLabel(ctx, "review/source-branch", req, client)
+	}
+	return !isMasterHeadBranch(req)
+}
+
+func validateBaseBranch(ctx context.Context, req types.PullRequestOuter, client *github.Client) bool {
+	if !isAgainstDefaultBranch(req) {
+		body := "Thank you for your contribution. It appears that you are submitting changes not against the default repository branch." +
+			"Please raise a new pull request againgst the default branch. "
+
+		comment(ctx, body, req, client)
+
+		applyLabel(ctx, "review/target-branch", req, client)
+	}
+
+	return isAgainstDefaultBranch(req)
+}
+
+func isValidHeadAndBaseBranch(req types.PullRequestOuter) bool {
+	return !isMasterHeadBranch(req) && isAgainstDefaultBranch(req)
+}
+
+func getDefaultBranch(req types.PullRequestOuter) string {
```

Comment:
How about attaching the methods to the type `types.PullRequestOuter`?

Such that you get: `req.getDefaultBranch()` rather than `getDefaultBranch(req)`? It should provide a more fluid / Go-like implementation.